### ISSUE TITLE
yubikey-manager CI breakage / updates required

### ISSUE THREAD

**Issue by grembo** [Yubico/yubikey-manager#507]
## TL;DR
- Update black to 22.3.0
- Update pyinstaller to 4.10

## Long Version
The yubikey-manager build pipeline breaks on Ubuntu and Windows, see #504 for an example.

### Ubuntu
Ubuntu build fails due to the version of `black` used not being compatible with the version of `click` used in the build pipeline, see:
- https://github.com/psf/black/issues/2964
- https://github.com/pallets/click/issues/2225

```
Traceback (most recent call last):
  File "/home/runner/.cache/pre-commit/repoj1p7t3i8/py_env-python3/bin/black", line 8, in <module>
    sys.exit(patched_main())
  File "/home/runner/.cache/pre-commit/repoj1p7t3i8/py_env-python3/lib/python3.10/site-packages/black/__init__.py", line 6606, in patched_main
    patch_click()
  File "/home/runner/.cache/pre-commit/repoj1p7t3i8/py_env-python3/lib/python3.10/site-packages/black/__init__.py", line 6[59](https://github.com/Yubico/yubikey-manager/runs/6550134337?check_suite_focus=true#step:5:60)5, in patch_click
    from click import _unicodefun  # type: ignore
ImportError: cannot import name '_unicodefun' from 'click' (/home/runner/.cache/pre-commit/repoj1p7t3i8/py_env-python3/lib/python3.10/site-packages/click/__init__.py)
```

### Windows
Windows build fails due to some issue with PyInstaller, which looks a lot like https://github.com/pyinstaller/pyinstaller/issues/6704

Updating pyinstaller to 4.10 should fix this problem.

```
Traceback (most recent call last):
    File "C:\Users\runneradmin\AppData\Local\pypoetry\Cache\virtualenvs\yubikey-manager-zH_Qgqg_-py3.10\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 363, in <module>
      main()
    File "C:\Users\runneradmin\AppData\Local\pypoetry\Cache\virtualenvs\yubikey-manager-zH_Qgqg_-py3.10\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 345, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "C:\Users\runneradmin\AppData\Local\pypoetry\Cache\virtualenvs\yubikey-manager-zH_Qgqg_-py3.10\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 164, in prepare_metadata_for_build_wheel
      return hook(metadata_directory, config_settings)
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-5g71kp9x\overlay\Lib\site-packages\setuptools\build_meta.py", line 188, in prepare_metadata_for_build_wheel
      self.run_setup()
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-5g71kp9x\overlay\Lib\site-packages\setuptools\build_meta.py", line 281, in run_setup
      super(_BuildMetaLegacyBackend,
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-5g71kp9x\overlay\Lib\site-packages\setuptools\build_meta.py", line 174, in run_setup
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 247, in <module>
      setup(
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-5g71kp9x\overlay\Lib\site-packages\setuptools\__init__.py", line 87, in setup
      return distutils.core.setup(**attrs)
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-5g71kp9x\overlay\Lib\site-packages\setuptools\_distutils\core.py", line 122, in setup
      dist.parse_config_files()
    File "C:\Users\runneradmin\AppData\Local\pypoetry\Cache\virtualenvs\yubikey-manager-zH_Qgqg_-py3.10\lib\site-packages\_virtualenv.py", line 21, in parse_config_files
      result = old_parse_config_files(self, *args, **kwargs)
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-5g71kp9x\overlay\Lib\site-packages\setuptools\dist.py", line 865, in parse_config_files
      setupcfg.parse_configuration(
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-5g71kp9x\overlay\Lib\site-packages\setuptools\config\setupcfg.py", line 168, in parse_configuration
      meta.parse()
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-5g71kp9x\overlay\Lib\site-packages\setuptools\config\setupcfg.py", line 447, in parse
      section_parser_method(section_options)
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-5g71kp9x\overlay\Lib\site-packages\setuptools\config\setupcfg.py", line [418](https://github.com/Yubico/yubikey-manager/runs/6550134442?check_suite_focus=true#step:10:419), in parse_section
      self[name] = value
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-5g71kp9x\overlay\Lib\site-packages\setuptools\config\setupcfg.py", line 239, in __setitem__
      value = parser(value)
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-5g71kp9x\overlay\Lib\site-packages\setuptools\config\setupcfg.py", line 553, in _parse_version
      return expand.version(self._parse_attr(value, self.package_dir, self.root_dir))
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-5g71kp9x\overlay\Lib\site-packages\setuptools\config\setupcfg.py", line 373, in _parse_attr
      return expand.read_attr(attr_desc, package_dir, root_dir)
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-5g71kp9x\overlay\Lib\site-packages\setuptools\config\expand.py", line 194, in read_attr
      module = _load_spec(spec, module_name)
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-5g71kp9x\overlay\Lib\site-packages\setuptools\config\expand.py", line 214, in _load_spec
      spec.loader.exec_module(module)  # type: ignore
    File "<frozen importlib._bootstrap_external>", line 883, in exec_module
    File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
    File "D:\a\yubikey-manager\yubikey-manager\pyinstaller\PyInstaller.py", line 15, in <module>
      from PyInstaller.__main__ import run
  ModuleNotFoundError: No module named 'PyInstaller.__main__'; 'PyInstaller' is not a package
  [end of output]
```

**Comment by dainnilsson** [comment]
CI has now been fixed, thanks for the detailed report!
